app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -683,26 +680,45 @@ void loadFromIntent(final Intent intent) {
             }
         }
 
-        Uri uri = intent.getData();
-
         boolean openInWindow = false;
         boolean openInBackground = false;
+        boolean openInKioskMode = false;
+
+        Uri dataUri = intent.getData();
+        Uri targetUri = null;
+        Bundle extras;
+
+        if (dataUri != null && dataUri.getScheme().equals("wolvic") && dataUri.getHost().equals("com.igalia.wolvic")) {
+            Log.d(LOGTAG, "Parsing custom URI from intent: " + dataUri);
+
+            extras = new Bundle();
+            Set<String> keys = dataUri.getQueryParameterNames();
+            for (String key : keys) {
+                if (key.equals("url") || key.equals("homepage"))
+                    extras.putString(key, dataUri.getQueryParameter(key));
+                else
+                    extras.putBoolean(key, Boolean.parseBoolean(dataUri.getQueryParameter(key)));
```

Comment:
I guess we cannot be sure that keys are either a string or a boolean. I am not thinking only in mischievous data being sent to Wolvic but also just on plain mistakes. I think we need this to be more robust.

Reply:
I think this is not too bad? The query parameters are always strings. `Boolean.parseBoolean()` simply checks that a string matches `"true"`.

Nevertheless, I have added a null check and replaced `equals()` with `equalsIgnoreCase()`.